optimum/habana/transformers/modeling_utils.py

```diff
@@ -356,6 +356,7 @@ def adapt_transformers_to_gaudi():
     transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXModel.forward = gaudi_gpt_neox_model_forward
     transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXLayer.forward = gaudi_gpt_neox_layer_forward
     transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXAttention.forward = gaudi_gpt_neox_attention_forward
+    transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXSdpaAttention.forward = gaudi_gpt_neox_attention_forward
```

Comment:
Not a big fan of doing this. This change should be enough: https://github.com/huggingface/optimum-habana/pull/1163/files#diff-bae02284f455b93397dbeafee178bd779671429602246f3ba60ea833a538eb68R297
Unless the test doesn't use `GPTNeoXForCausalLM`? Which test is it?

Reply:
Just saw the test above, let me check

Reply:
I pushed this commit to do it differently, the test above still passes